full-stack-tests/presentation/src/learning-snippets/Content.test.ts

```diff
@@ -601,115 +601,8 @@ describe("Learning Snippets", () => {
-        const ruleset: Ruleset = {
-          id: "example",
-          rules: [{
-            ruleType: RuleTypes.Content,
-            specifications: [{
-              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
-              classes: { schemaName: "BisCore", classNames: ["PhysicalModel"], arePolymorphic: false },
-            }],
-          }],
-        };
-        // __PUBLISH_EXTRACT_END__
-
-        // Ensure only the `bis.PhysicalModel` instances are selected.
-        const content = await Presentation.presentation.getContent({
-          imodel,
-          rulesetOrId: ruleset,
-          keys: new KeySet(),
-          descriptor: {},
-        });
-
-        expect(content!.contentSet.length).to.eq(1);
-        expect(content!.contentSet[0].primaryKeys[0].className).to.eq("BisCore:PhysicalModel");
-      });
-
-      it("uses `handlePropertiesPolymorphically` attribute", async () => {
-        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.HandlePropertiesPolymorphically.Ruleset
-        // The specification returns content of all `bis.ViewDefinition` instances
-        // with properties of all `bis.ViewDefinition` subclasses.
-        const ruleset: Ruleset = {
-          id: "example",
-          rules: [{
-            ruleType: RuleTypes.Content,
-            specifications: [{
-              specType: ContentSpecificationTypes.ContentInstancesOfSpecificClasses,
-              classes: { schemaName: "BisCore", classNames: ["ViewDefinition"], arePolymorphic: true },
-              handlePropertiesPolymorphically: true,
-            }],
-          }],
-        };
-        // __PUBLISH_EXTRACT_END__
-
-        // Ensure that derived `bis.ViewDefinition` instances along with their properties are also selected.
-        const content = await Presentation.presentation.getContent({
-          imodel,
-          rulesetOrId: ruleset,
-          keys: new KeySet(),
-          descriptor: {},
-        });
-        expect(content!.descriptor.fields).to.containSubset([
-          { label: "Category Selector" },
-          { label: "Code" },
-          { label: "Description" },
-          { label: "Display Style" },
-          { label: "Extents" },
-          { label: "Eye Point" },
-          { label: "Focus Distance" },
-          { label: "Is Camera On" },
-          { label: "Is Private" },
-          { label: "Lens Angle" },
-          { label: "Model" },
-          { label: "Model Selector" },
-          { label: "Origin" },
-          { label: "Pitch" },
-          { label: "Roll" },
-          { label: "User Label" },
-          { label: "Yaw" },
-        ]).and.to.have.lengthOf(17);
-
-        expect(content!.contentSet.length).to.eq(4);
-      });
-
       it("uses `instanceFilter` attribute", async () => {
-        // __PUBLISH_EXTRACT_START__ ContentInstancesOfSpecificClasses.InstanceFilter.Ruleset
+        // __PUBLISH_EXTRACT_START__ SharedAttributes.InstanceFilter.Ruleset
```

Comment:
I don't like how these common attributes only use `ContentInstancesOfSpecificClasses` specification, but appear under other specifications too. For example, here it's really not clear whether this filter is applied to the source instance, or instances that are discovered through the relationships.

Reply:
Having put more thought, perhaps it's worth making a new example for this case since the filter does apply to the `relationshipPath` property and not `classes`.